llvm/lib/SYCLLowerIR/SYCLPropagateAspectsUsage.cpp

```diff
@@ -655,16 +687,21 @@ buildFunctionsToAspectsMap(Module &M, TypeToAspectsMapTy &TypesWithAspects,
                            bool ValidateAspects, bool FP64ConvEmu) {
   FunctionToAspectsMapTy FunctionToUsedAspects;
   FunctionToAspectsMapTy FunctionToDeclaredAspects;
+  StringMap<std::vector<Function *>> VirtualFunctionSets;
```

Comment:
Any reason we are using `std::vector` over `SmallVector` or some other LLVM data structure?
Also we might want to make a type alias so the type is cleaner.

Reply:
It should likely be `SmallVector`. At least for sequential containers [LLVM Programmer's Manual](https://llvm.org/docs/ProgrammersManual.html#sequential-containers-std-vector-std-list-etc) is clear that you should pick the first one which suites your needs and `std::vector` is at the very end of the list